client/types.go

```diff
@@ -1,6 +1,33 @@
 package client
 
-import "time"
+import (
+	"fmt"
+	"strings"
+	"time"
+)
 
 // DateTime is a bridge between a swagger doc "date-time" and a Golang time.Time type
-type DateTime = time.Time
+type DateTime struct {
+	time.Time
+}
+
+func (dt *DateTime) UnmarshalJSON(b []byte) error {
+	s := strings.Trim(string(b), "\"")
+	if s == "null" {
+		dt.Time = time.Time{}
+		return nil
+	}
+	parsedTime, err := time.Parse(time.DateOnly, s)
```

Comment:
Isn't using only `time.DateOnly` pretty rigid? Right now I believe we only use this for PR "days" but in the future, if we use other "date-time" formats, which should follow the [very general internet date/time format](https://datatracker.ietf.org/doc/html/rfc3339#section-5.6), this will force any `date-time` to always just be "days"?

Reply:
The parsing that I implemented doesn't account for time. I should have renamed the type to `Date` here.
If we go forward with the patching, I think it might be better to have 2 separate struct types - `Date` & `DateTime`.